_examples/real-world-examples/server-sent-events/server/event_handlers.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/message/router/middleware"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+)
+
+const (
+	PostCreatedTopic = "post-created"
+	PostUpdatedTopic = "post-updated"
+	FeedUpdatedTopic = "feed-updated"
+)
+
+func SetupMessageRouter(
+	feedsStorage FeedsStorage,
+	logger watermill.LoggerAdapter,
+) (message.Publisher, message.Subscriber, error) {
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	router.AddMiddleware(middleware.Recoverer)
+
+	pubsub := gochannel.NewGoChannel(gochannel.Config{}, logger)
+
+	publishEvents := func(ctx context.Context, tags []string) (messages []*message.Message, err error) {
```

Comment:
why not create a normal function, instead of a variable?

Reply:
also, this does not seem to publish any messages, just return them - so the name doesn't fit

Reply:
Seems like a quick idea I did one year ago and didn't refactor properly. 😅 

c4eba92ae51e6cb3656a235ee014acfb11293212